src/main/scala/TextToStructuredData.scala

```diff
@@ -0,0 +1,25 @@
+import org.apache.spark.sql.functions._
+import TweetsNormalization.normalizeTweets
+import FeaturesExtractionFromRawTweet.extractFeaturesFromRawTweet
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+object TextToStructuredData extends SparkSessionCreator {
+  def getStructuredDataFromText(text: Dataset[String]): DataFrame = {
```

Comment:
you have defined a data model / schema for a tweet in `StructuredTweet` class, but the API performing the extraction doesn't utilizes it (i.e. returns `DataFrame`, not `StructuredTweet`). 

If the function does something else or more than it's expected to do (just building structured data from plan text) then it takes too much responsibility (i.e. incoherent).
Or it could that the `StructuredTweet` is not defined well enough to represent all required fields, so additional high-level model of `Row` is used here. In this case, `StructuredTweet` should just include the missing fields.

The goal is to make API utilize the defined model, so it's clear for consumers (at compile time) what data schema is.

Reply:
Change returned type to Dataset[StructuredData]. This method take Dataset[String] as argument and concat some lines of it at first. So it's cannot be implemented as String to StructuredData method (many to one relationship).